amazon/ion/writer_text.py

```diff
@@ -173,15 +173,33 @@ def _bytes_datetime(dt):
         return tz_string + _bytes_utc_offset(dt)
 
     fractional_precision = getattr(original_dt, TIMESTAMP_FRACTION_PRECISION_FIELD, MICROSECOND_PRECISION)
-    if fractional_precision:
+    fractional_seconds = getattr(original_dt, TIMESTAMP_FRACTIONAL_SECONDS_FIELD, None)
+    if fractional_precision is not None and fractional_seconds is None:
         fractional = dt.strftime('%f')
         assert len(fractional) == MICROSECOND_PRECISION
+
         if fractional[fractional_precision:] != ('0' * (MICROSECOND_PRECISION - fractional_precision)):
             raise ValueError('Found timestamp fractional with more than the specified %d digits of precision.'
                              % (fractional_precision,))
         fractional = fractional[:fractional_precision]
         tz_string += '.' + fractional
-
+    else:
+        # Since 0.000... is interpreted as 0, must do this in order to have full precision 0's.
+        if fractional_seconds == 0:
+            tz_string += '.' + ('0' * fractional_precision)
+        elif fractional_seconds is not None:
+            if 'e' in str(fractional_seconds):
+                fractional_string = str(Decimal(fractional_seconds))[1:]
```

Comment:
Why `str(Decimal(fractional_seconds))` and not just `str(fractional_seconds)`?

Where do we avoid writing the 'e'?

Does it make more sense (and be cleaner) to just write the coefficient scaled by -exponent (which must be an integer)?

Reply:
This logic with the 'e' was not necessary, and it has been removed. I did not find it necessary to write the coefficient scaled by the exponent, as `fractional_seconds` is always a Decimal? Please review.

Reply:
Let's continue this discussion on my latest review.